domain/src/test/java/org/oppia/android/domain/classify/rules/fractioninput/FractionInputIsGreaterThanRuleClassifierProviderTest.kt

```diff
@@ -0,0 +1,447 @@
+      inputs = inputs
+    )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testNegativeFractionalAnswer_negativeFractionalInput_answerGreaterThan_verifyAnswerGreater() {
+    val inputs = mapOf("f" to NEGATIVE_FRACTION_1_OVER_2)
+
+    val matches = inputGreaterThanRuleClassifier.matches(
+      answer = NEGATIVE_FRACTION_1_OVER_3,
+      inputs = inputs
+    )
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testNegativeFractionalAnswer_negativeFractionalInput_answerSmallerThan_verifyAnswerSmaller() {
+    val inputs = mapOf("f" to NEGATIVE_FRACTION_1_OVER_3)
+
+    val matches = inputGreaterThanRuleClassifier.matches(
+      answer = NEGATIVE_FRACTION_1_OVER_2,
+      inputs = inputs
+    )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testNegativeFractionalAnswer_positiveFractionalInput_answerSmallerThan_verifyAnswerSmaller() {
+    val inputs = mapOf("f" to FRACTION_1_OVER_2)
+
+    val matches = inputGreaterThanRuleClassifier.matches(
+      answer = NEGATIVE_FRACTION_1_OVER_3,
+      inputs = inputs
+    )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testNegativeMixedAnswer_negativeMixedInput_exactSameValue_verifyAnswerSmaller() {
+    val inputs = mapOf("f" to NEGATIVE_MIXED_NUMBER_123_1_OVER_2)
+
+    val matches = inputGreaterThanRuleClassifier.matches(
+      answer = NEGATIVE_MIXED_NUMBER_123_1_OVER_2,
+      inputs = inputs
+    )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testNegativeMixedAnswer_negativeMixedInput_answerGreaterThan_verifyAnswerGreater() {
+    val inputs = mapOf("f" to NEGATIVE_MIXED_NUMBER_123_1_OVER_2)
+
+    val matches = inputGreaterThanRuleClassifier.matches(
+      answer = NEGATIVE_MIXED_NUMBER_123_1_OVER_3,
+      inputs = inputs
+    )
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testNegativeMixedAnswer_negativeMixedInput_answerSmallerThan_verifyAnswerSmaller() {
+    val inputs = mapOf("f" to NEGATIVE_MIXED_NUMBER_123_1_OVER_3)
+
+    val matches = inputGreaterThanRuleClassifier.matches(
+      answer = NEGATIVE_MIXED_NUMBER_123_1_OVER_2,
+      inputs = inputs
+    )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testNegativeMixedAnswer_positiveMixedInput_answerSmallerThan_verifyAnswerSmaller() {
+    val inputs = mapOf("f" to MIXED_NUMBER_123_1_OVER_2)
+
+    val matches = inputGreaterThanRuleClassifier.matches(
+      answer = NEGATIVE_MIXED_NUMBER_123_1_OVER_2,
+      inputs = inputs
+    )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testNegativeMixedAnswer_negativeFractionalInput_answerSmallerThan_verifyAnswerSmaller() {
+    val inputs = mapOf("f" to NEGATIVE_FRACTION_1_OVER_3)
+
+    val matches = inputGreaterThanRuleClassifier.matches(
+      answer = NEGATIVE_MIXED_NUMBER_123_1_OVER_2,
+      inputs = inputs
+    )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testNegativeFractionalAnswer_negativeMixedInput_answerGreaterThan_verifyAnswerGreater() {
```

Comment:
Put all `testNegativeFractionalAnswer` cases together. When we write ditto, it applies to all the cases in the file. Here, you've jumbled up between `testNegativeMixedAnswer` and `testNegativeFractionalAnswer`.

Reply:
done